api/src/main/java/org/jboss/marshalling/ClassNameTransformer.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2021 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.marshalling;
+
+/**
+ * A class name transformer. Allows to remap one java type name to another java type name.
+ *
+ * @author <a href="mailto:ropalka@redhat.com">Richard Opalka</a>
+ */
+public interface ClassNameTransformer {
+
+    /**
+     * Transforms unmarshaller input stream java type names.
+     * @param className candidate to be translated
+     * @return either original or new class name
+     */
+    String transformInput(String className);
+
+    /**
+     * Transforms marshaller output stream java type names.
+     * @param className candidate to be translated
+     * @return either original or new class name
+     */
+    String transformOutput(String className);
+
+    /**
+     * Translates all incoming javax EE types to jakarta EE types and all outgoing jakarta EE types to javax EE types.
```

Comment:
Is the list of class transformations likely to evolve? The documentation should say what Java EE version and what Jakara version are assumed.

If you are using the word "Jakarta" to refer to the specification, then it should be spelled with an uppercase "J" like "Jakarta" or "Jakarta EE"; but if you are referring to the package name, you should specify it like this: `{@code jakarta}` (and likewise: `{@code javax}`). "Java EE" should be "Java EE", not "javax EE" (the "x" isn't part of the specification name and the "EE" is not part of the package name).

Reply:
Fixed.